File: pset6/credit/credit.py
from cs50 import get_int,get_string
from sys import exit

x=get_string("Number: ")
len_card=len(x)
if len_card <13 or len_card> 16:
    print("INVALID")
    exit(1)
checksum=0


for i in range(len_card-2,-1,-2):
    if(int(x[i]) * 2 > 9):
        checksum += (int(x[i]) *2 ) -9
    else:
        checksum += int(x[i]) * 2

for k in range(len_card-1,-1,-2):
    checksum += int(x[k])


if checksum%10 == 0:
    if x[0:2] == "34" or x[0:2] == "37" and len_card == 15:
        print("AMEX")
    elif x[0:2] in ["51","52","53","54","55"]:
        print("MASTERCARD")
    elif x[0] == "4" and len_card == 13 or len_card == 16:
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")

File: PSET1/credit.c
#import <cs50.h>
#import <stdio.h>


int main(void)
{
int vetor[16],sum1=0,sum2=0,checksum;
long card;
int mod,cont=0;
do{
card=get_long("Number: ");
}
while(card<0);
for(int i=0;card>0;i++)
{
    if(cont>15)
    {
        printf("INVALID\n");
        return 0;
    } 
    mod=card%10;
    vetor[i]=mod;
    card/=10;
    cont++;
}
if(cont<13)
{
    printf("INVALID\n");
    return 0;
}
for(int i=1;i<cont;i+=2)
{
if((vetor[i]*2)>9)
{
    sum1+=(vetor[i]*2)-9;
}
else
{
   sum1+=vetor[i]*2;
}
}
for(int i=0;i<cont;i+=2)
{
    sum2+=vetor[i];
}
checksum=sum1+sum2;
if(checksum%10 ==0)
{
    //printf("Valid\n");
    if((vetor[cont-1]==3&&vetor[cont-2]==4|vetor[cont-1]==3&&vetor[cont-2]==7)&& cont==15)
    {
        printf("AMEX\n");
    }
    else if((vetor[cont-1]==4)&&cont==13|cont==16)
    {
        printf("VISA\n");
    }
    else if(vetor[cont-1]==5&&vetor[cont-2]==1|vetor[cont-2]==2|vetor[cont-2]==3|vetor[cont-2]==4|vetor[cont-2]==5)
    {
        printf("MASTERCARD\n");
    }
    else
    {
        printf("INVALID\n");
    }
}
else
{
    printf("INVALID\n");
}
}
